docs/user_guide/index.rst

```diff
@@ -533,3 +533,66 @@ Then, follow the pattern of normal startup but specify your function in the :py:
 
 The benefit of this approach over manipulating the frame later is that the Seismogram can be considered frozen.
 Among other things, this means any Seismograph notebooks cells do not have a dependence on order and can be run multiple times.
+
+
+Create Custom Visualizations
+----------------------------
+
+You can create custom controls that allow users to interact with the data via a set of 
+standardized controls. The :py:mod:`seismometer.controls.explore` module contains several ``Exploration*``
+widgets you can use for housing custom visualizations, see :ref:`custom-visualization-controls`. 
+
+.. image:: media/custom_plot_control.png
+   :alt: A custom control, allowing a user to select a cohort and display a heatmap restricted to that cohort.
+   :width: 4.5in
+
+To add your own custom visualization, you need a function that takes the same signature as the Exploration widget, and
+it should return a displayable object. If using matplotlib, you can use the :py:func:`~seismometer.plot.mpl.decorators.render_as_svg`
+to convert the plot to an SVG, for the control to display.
+This will close the plot/figure after saving to prevent the plot from displaying twice.
+
+The following example shows how to create a visualization above. 
+
+.. code-block:: python
+
+   import seaborn as sns
+   import matplotlib.pyplot as plt
+
+   # Control allowing users to specify a score, target, threshold, and cohort. 
+   from seismometer.controls.explore import ExplorationModelSubgroupEvaluationWidget 
+   # Converts matplotlib figure to SVG for display within the control's output
+   from seismometer.plot.mpl.decorators import render_as_svg
+   # Filter our data based on a specified cohort 
+   from seismometer.data.filter import FilterRule 
+
+   @render_as_svg
+   def plot_heat_map(
+         cohort_dict: dict[str,tuple], # cohort columns and allowable values
+         target_col: str, # the model target column
+         score_col: str,  # the model output column (score)
+         thresholds: tuple[float], # a list of thresholds to consider
+         *,
+         per_context: bool # if a plot groups scores by context
+         ) -> plt.Figure:
+      xcol = "age"
+      ycol = "num_procedures"
+      hue = score_col
+
+      sg = sm.Seismogram()
```

Comment:
It's probably worth while to split the cohort_filter.filter() and pandas slicing (line~584), that way we can call out these "three" lines (582-584) are going to be useful in their plot exactly.

Reply:
FINC